gentle_gnomes/src/azavea.py

```diff
@@ -57,18 +62,65 @@ def get_nearest_city(self, lat: float, lon: float, limit: int = 1, **kwargs) ->
             city = cities['features'][0]
             return City(city['properties']['name'], city['properties']['admin'], city['id'])
 
-    def get_scenarios(self, **kwargs) -> List:
+    def get_scenarios(self, **kwargs) -> t.List:
         """Return all available scenarios."""
         return self._get('/scenario', **kwargs)
 
-    def get_indicators(self, **kwargs) -> Dict:
+    def get_indicators(self, **kwargs) -> t.Dict:
         """Return the full list of indicators."""
         return self._get('/indicator', **kwargs)
 
-    def get_indicator_details(self, indicator: str, **kwargs) -> Dict:
+    def get_indicator_details(self, indicator: str, **kwargs) -> t.Dict:
         """Return the description and parameters of a specified indicator."""
         return self._get(f'/indicator/{indicator}', **kwargs)
 
-    def get_indicator_data(self, city: int, scenario: str, indicator: str, **kwargs) -> Dict:
+    def get_indicator_data(self, city: int, scenario: str, indicator: str, **kwargs) -> t.Dict:
         """Return derived climate indicator data for the requested indicator."""
         return self._get(f'/climate-data/{city}/{scenario}/indicator/{indicator}', **kwargs)
+
+
+class AsyncProxy:
+
+    def __init__(self):
+        self._pending = []  # [(url, kwargs), ...]
+
+    async def _process_requests(self, *requests: t.Tuple[str, dict]) -> t.Tuple[t.Union[dict, list]]:
+        session = aiohttp.ClientSession()
```

Comment:
I think it's better to keep re-using a session rather than making a new one. Can be moved to `__init__` and let that take  `**kwargs` to pass to the session, including the header.

Reply:
That is true, but aiohttp complains when we instantiate the session outside of an async loop.

I agree the default header is a little funky, maybe there is a better way to keep that session open.

Reply:
[Here](https://github.com/aio-libs/aiohttp/pull/1468) is the issue where they discuss some solutions, like an async context manager or some method called after instantiation which creates the session. They made the warning disappear when `loop` was specified but that parameter [has been deprecated](https://github.com/aio-libs/aiohttp/issues/3331). Basically they want the session to be created within an asyncio loop.

Reply:
Since our entire program isn't async I suppose what we can do is create the session in this method if it doesn't exist or otherwise re-use it. But we'd need to think of a way to properly close the session still.

Reply:
I take that back, I'd rather not do a lazy-load like this. Better to do it when class is instantiated. Fuck

Reply:
I'm not sure the lazy load is avoidable. Unless we make `create_app` async, lazy loading seems like the best way to do it.

Reply:
Opening an issue, this needs some more thought.